src/node/kernel_notifications.cpp

```diff
@@ -99,6 +100,14 @@ void KernelNotifications::fatalError(const bilingual_str& message)
                     m_exit_status, message, &m_warnings);
 }
 
+std::optional<uint256> KernelNotifications::TipBlock()
+{
+    AssertLockHeld(m_tip_block_mutex);
+    Assert(!m_tip_block || m_tip_block.value() != uint256{});
```

Comment:
we're already doing validation during assignment, this seems redundant to me

Reply:
It is now, but maybe we introduce assignment somewhere else and forget to check it there. This would catch such a mistake.

It's a bit overkill, which we could drop once the whole codebase uses a single pattern.

Reply:
> It's a bit overkill, which we could drop once the whole codebase uses a single pattern.

Valid argument